### ISSUE TITLE
How flexible TimeSformer model is for tiny dataset(1 or 2 training videos)

### ISSUE THREAD

**Issue by Joeycho** [facebookresearch/TimeSformer#51]
Hello,

In principle, since TimeSformer has a large number of parameter to learn dataset, I assume that learning one or two training videos should be possible with TimeSformer. (Intend to overfit on the training set, for debugging purpose)

However, compared to 300 training videos, training on two videos seems not possible with TimeSformer. That's how I ask how flexible TimeSformer as  model. The learning capacity is large, but not able to learn 2 training videos? Or maybe this is because of pretrained weights from ViT and ImageNET?

Or I got the wrong result with one training video?

I could see that the Accuracy gets higher along the number of training videos in Paper. But now, I am talking about Training loss and Training Accuracy, not with Test set. 
 
Sky color line depicts two training videos, and blue line depicts 300 training videos. 
 
![with_one_sample_compared_to_300](https://user-images.githubusercontent.com/29337166/123244741-7b3a9180-d4e4-11eb-942a-f820b09dfa01.JPG)

**Comment by gberta** [comment]
Training TimeSformer on 1-2 videos is a strange setup. I'm not sure what your rationale for doing this is. I haven't tried it myself, so it's difficult to comment. However, in theory, I don't see any reason why the model wouldn't be able to overfit on 1-2 videos. Therefore, I suspect that there's a bug somewhere in your setup. How many GPUs are you using to train your model?

Overall, I think it would be more useful if you told me what you are trying to do. I don't think training on 1-2 videos makes much sense.

**Comment by Joeycho** [comment]
@gberta I am ultimately trying to retrain TimeSformer on HVU Dataset. Ant the reason for overfitting on 1-2 videos is for debugging purpose before moving on more videos(~480K videos). I see. Then I would try to fix and figure it out why the model was not able to overfit on 1-2 videos. So, I understood it as 'debug signal'. I don't really try to do it for itself.

**Comment by Joeycho** [comment]
It was successfully overfitted with the constant learning rate. Loss decreased with oscillation.
